### PR TITLE
Implement enum support in feature search & query builder

### PR DESCRIPTION
**Pull request by KyleJu**
#2344.

Server side:
- When parsing a query term on an enum field, check if the value is a string that is one of the enum choices. If so, convert it to the appropriate integer. If it is not one of the enum choices, treat that as an error.

Client side:
- Add sl-select widget to offer the user choices of enum values
- Generate a query term like field_name="enum value string" for enum fields

### CODE REVIEW

File: client-src/elements/chromedash-feature-filter.js [hunk 1]
```diff
@@ -144,6 +156,57 @@ const QUERIABLE_FIELDS = [
...
+    type: ENUM_TYPE},
+  {name: 'privacy_review_status',
+    display: 'Privacy review status',
+    choices: REVIEW_STATUS_CHOICES,
+    type: ENUM_TYPE},
+  {name: 'tag_review_status',
+    display: 'TAG Specification Review Status',
+    choices: REVIEW_STATUS_CHOICES,
+    type: ENUM_TYPE},
+  {name: 'safari_views',
```
Comment (jrobbins) [inline client-src/elements/chromedash-feature-filter.js:198]: These should use the dotted property names like the ones commented out on lines 123-128.   And then those commented out lines can be deleted.

File: internals/core_enums.py [hunk 2]
```diff
@@ -506,3 +506,17 @@ def convert_enum_int_to_string(property_name, value):
...
+  """If the property is an enum, return its enum value, else -1."""
+  enum_dict = PROPERTY_NAMES_TO_ENUM_DICTS.get(property_name, {})
+  for index, enum_str in enumerate(enum_dict):
+    if enum_str == value:
+      return index
+  return -1
+
+
+def is_enum_field(property_name):
+  enum_dict = PROPERTY_NAMES_TO_ENUM_DICTS.get(property_name, {})
```
Comment (jrobbins) [inline internals/core_enums.py:521]: This could be: `return property_name in PROPERTY_NAMES_TO_ENUM_DICTS`

File: internals/search_queries.py [hunk 3]
```diff
@@ -257,6 +256,20 @@ def sorted_by_review_date(descending: bool) -> list[int]:
...
+    # Enum fields
+    'category': FeatureEntry.category,
+    'intent_stage': FeatureEntry.intent_stage,
+    'impl_status_chrome': FeatureEntry.impl_status_chrome,
+    'security_review_status': FeatureEntry.security_review_status,
+    'privacy_review_status': FeatureEntry.privacy_review_status,
+    'standards.maturity': FeatureEntry.standard_maturity,
+    'standardization': FeatureEntry.standardization,
+    'browsers.ff.view': FeatureEntry.ff_views,
+    'ie_views': FeatureEntry.ie_views,
```
Comment (jrobbins) [inline internals/search_queries.py:269]: Likewise, dotted names for these.

**Comment by KyleJu** [comment]
> There's a test failure referencing `standardization`. That field was deprecated in Feature so I did not add it to FeatureEntry. Instead there is `standard_maturity`.

'rollout_platforms' seems to be a field in stage. I will add it separately as it is not supported yet

**Comment by KyleJu** [comment]
Seeing a bug for "feature type" search. I will send out a fix in a follow-up PR. Otherwise this PR has been tested on staging https://33c0dce-dot-cr-status-staging.appspot.com/newfeatures

**Pull request merged**